vars/neoDeploy.groovy

```diff
@@ -1,94 +1,85 @@
         }
 
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         if(defaultDeployAccount) {
-            echo "[WARNING][${stepName}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_ACCOUNT'. This configuration framekwork will be removed in future versions."
-            stepConfiguration.put('account', defaultDeployAccount)
+            echo "[WARNING][${STEP_NAME}] A deprecated configuration framework is used for configuring parameter 'DEPLOY_ACCOUNT'. This configuration framekwork will be removed in future versions."
+            stepCompatibilityConfiguration.put('account', defaultDeployAccount)
         }
 
         if(parameters.deployHost && !parameters.host) {
-            echo "[WARNING][${stepName}] Deprecated parameter 'deployHost' is used. This will not work anymore in future versions. Use parameter 'host' instead."
+            echo "[WARNING][${STEP_NAME}] Deprecated parameter 'deployHost' is used. This will not work anymore in future versions. Use parameter 'host' instead."
             parameters.put('host', parameters.deployHost)
         }
 
         if(parameters.deployAccount && !parameters.account) {
-            echo "[WARNING][${stepName}] Deprecated parameter 'deployAccount' is used. This will not work anymore in future versions. Use parameter 'account' instead."
+            echo "[WARNING][${STEP_NAME}] Deprecated parameter 'deployAccount' is used. This will not work anymore in future versions. Use parameter 'account' instead."
             parameters.put('account', parameters.deployAccount)
         }
 
         def credId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
-
         if(credId && !parameters.neoCredentialsId) {
-            echo "[WARNING][${stepName}] Deprecated parameter 'neoCredentialsId' from old configuration framework is used. This will not work anymore in future versions."
+            echo "[WARNING][${STEP_NAME}] Deprecated parameter 'neoCredentialsId' from old configuration framework is used. This will not work anymore in future versions."
             parameters.put('neoCredentialsId', credId)
         }
-
         // Backward compatibility end
-
-        stepConfiguration.putAll(ConfigurationLoader.stepConfiguration(script, stepName))
-
-        Map configuration = ConfigurationMerger.merge(parameters, parameterKeys,
-                                                      stepConfiguration, stepConfigurationKeys,
-                                                      ConfigurationLoader.defaultStepConfiguration(script, stepName))
+        
+        // load default & individual configuration
+        Map configuration = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixin(stepCompatibilityConfiguration)
```

Comment:
shouldn't we add this to the compatibility layer we already have in place?
Example: https://github.com/SAP/jenkins-library/blob/master/vars/artifactSetVersion.groovy#L47

We may add a default Deprecation message if required into the compatibility layer: https://github.com/SAP/jenkins-library/blob/master/src/com/sap/piper/ConfigurationHelper.groovy#L56